include/data/json.h

```diff
@@ -86,6 +86,7 @@ struct json_obj_descr {
 		struct {
 			const struct json_obj_descr *element_descr;
 			size_t n_elements;
+			uint8_t nested : 1;
```

Comment:
I agree it would be good to avoid an increase of `json_obj_descr` size.

Hoevever, creating another `JSON_TOK_ARRAY_NESTED` which would also indicate starting of an array will most probably not work, as the values behind the elements in `enum json_token` are used as characters for parsing in `json.c`.